geth/node/node.go

```diff
@@ -119,6 +119,11 @@ func defaultEmbeddedNodeConfig(config *params.NodeConfig) *node.Config {
 		nc.HTTPPort = config.HTTPPort
 	}
 
+	if config.WhisperConfig.Standalone {
```

Comment:
I think you should check for `nodeConfig.BootClusterConfig.Enabled`. Whisper config should not affect `node`.

Reply:
same as above